Scan.cpp

```diff
@@ -0,0 +1,99 @@
+#include "ProcessManager.h"
+
+#include <assert.h>
+
+#include <vector>
+#include <utility>
+
+using namespace std;
+
+/**
+ * Takes a maps file and returns a vector of pairs 
+ * having the beginning and ending addresses of readable
+ * memory locations.
+ * 
+ * @param fileBuffer the maps file for a process
+ * @returns a vector pairs
+ */
+vector<pair<long, long>> getRanges(char *fileBuffer){
+	vector<pair<long, long>> ranges;
+	char *p;
+	char *m = fileBuffer;
+	while((p=strstr(m, "\n"))!=NULL){
+		char *start = p-1;
+		while( (*start) != '\n' && start>=fileBuffer)
+		{
+			start--;
+		}
+		start++;
+		char *c=strstr(start, "---p");
+		if(c&&c<strstr(start,"\n")){
+			m=p+1;
+			continue;
+		}
+		char chAddress[1024];
+		assert(chAddress != NULL);
+		memset(chAddress, 0, 1024);
+		for(int i=0;*start!='-';i++)
+		{
+			chAddress[i] = *start;
+			start++;
+		}
+		pair<long, long> range;
+		range.first = strtol(chAddress, NULL, 16);
+		memset(chAddress, 0, 1024);
+		start++;
+		for (int i = 0; *start != ' '; i++)
+		{
+			chAddress[i] = *start;
+			start++;
+		}
+		range.second = strtol(chAddress, NULL, 16);
+		ranges.push_back(range);
+		m = p + 1;
+	}
+	return ranges;
+}
+
+/**
```

Comment:
change comment style in accordance with doxy (refer discord)

Reply:
done